addons/stock/models/res_company.py

```diff
@@ -126,12 +141,20 @@ def create_missing_scrap_location(self):
         for company in company_without_property:
             company._create_scrap_location()
 
+    @api.model
+    def create_missing_scrap_sequence(self):
+        company_ids  = self.env['res.company'].search([])
+        company_has_scrap_seq = self.env['ir.sequence'].search([('code', '=', 'stock.scrap')]).mapped('company_id')
+        company_todo_sequence = company_ids - company_has_scrap_seq
+        company_todo_sequence._create_scrap_sequence()
+
     def _create_per_company_locations(self):
         self.ensure_one()
         self._create_transit_location()
         self._create_inventory_loss_location()
         self._create_production_location()
         self._create_scrap_location()
+        self._create_scrap_sequence()
```

Comment:
not the right place

Reply:
maybe put it directly in _create_scrap_location, this way it is called when installing stock (the method is called in data.xml)

cc @amoyaux if he has a better idea